app/helpers/image_conversion_helper.rb

```diff
@@ -2,26 +2,80 @@ module ImageConversionHelper
   IMAGE_QUALITY = 50
 
   def resize_to_dimensions_and_convert_to_jpg(source_image_path:, attachment_type:, filename:)
-    # Set dimensions
-    mini_magick_image = MiniMagick::Image.open(source_image_path)
+    file = File.binread(source_image_path)
+    file_bytes = file.unpack("C*")
+
+    dimensions = nil
     if attachment_type == SiteBanner::LOGO
-      mini_magick_image.resize(SiteBanner::LOGO_DIMENSIONS.join("x"))
+      dimensions = SiteBanner::LOGO_DIMENSIONS
     elsif attachment_type == SiteBanner::BACKGROUND
-      mini_magick_image.resize(SiteBanner::BACKGROUND_DIMENSIONS.join("x"))
-    else
+      dimensions = SiteBanner::BACKGROUND_DIMENSIONS
+    end
+
+    if !dimensions
       LogException.perform(StandardError.new("Unknown attachment type:" + attachment_type), params: {})
       return nil
     end
 
-    # Use JPG as its the most efficient standard
-    mini_magick_image.format "jpg"
+    actual_jpeg_hex_to_save = resize_image_with_wasm(file_bytes: file_bytes, dimensions: dimensions)
 
     new_filename = filename + "_resized"
-    temp_file = Tempfile.new([new_filename, ".jpg"])
+    temp_file = Tempfile.new([new_filename, ".jpg"], binmode: true)
+    temp_file.write(actual_jpeg_hex_to_save)
+    temp_file.path
+  end
 
-    mini_magick_image.write(temp_file)
+  def resize_image_with_wasm(file_bytes:, dimensions:, size: 250000)
+    require "wasmer"
+    store = Wasmer::Store.new
+    # Let's compile the module to be able to execute it!
+    module_ = Wasmer::Module.new store, IO.read("#{Rails.root}/wasm_thumbnail.wasm", mode: "rb")
```

Comment:
I think we should make this a global object that's initialized in the initializers folder rather than creating a new object every time.

Reply:
better yet - let's create a wrapper gem that packages the wasm file and creates the global

Reply:
I've at least moved the compilation out to the module level. We can leave the gem for a later exercise.

Reply:
I ran into issues moving the instance itself into a global, weird cryptic ruby errors, still tracking down. For now, I've moved the compilation to the module level and keep the instance in the method.

Reply:
I've got a local rubygem, will push it up later.

Reply:
Gem here https://github.com/brave-intl/wasm-thumbnail/pull/2